src/WebJobs.Extensions.DurableTask/Listener/DurableTaskTriggerMetrics.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+#if NETSTANDARD2_0
+using System;
+using System.Collections.Generic;
+using System.Text;
+using DurableTask.AzureStorage.Monitoring;
+using Microsoft.Azure.WebJobs.Host.Scale;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal class DurableTaskTriggerMetrics : ScaleMetrics
+    {
+        /// <summary>
+        /// The number of partitions in the task hub.
+        /// </summary>
+        public int PartitionCount { get; set; }
+
+        /// <summary>
+        /// The number of messages across control queues.
+        /// </summary>
+        public string ControlQueueLengths { get; set; }
+
+        /// <summary>
+        /// The latency of messages across control queues.
+        /// </summary>
+        public string ControlQueueLatencies { get; set; }

```

Comment:
These strings seem to map to `List<int>`s. Is there a reason we have to store them as strings here?

Reply:
Yes -- I've included some examples in the comments (per Mathew) that should help. We have to serialize and deserialize these values because they're being stored in an Azure Storage table that can't handle complex objects

Reply:
Yeah, it's unfortunate we have to convert to string here. The underlying storage for metrics is pluggable, but the default impl. is Azure Tables, and it only supports simple properties (code [here](https://github.com/Azure/azure-functions-host/blob/643c21f882218283a894dc29d78ab4088e073de7/src/WebJobs.Script.WebHost/Scale/TableEntityConverter.cs#L76)). There isn't any built in conversions from unsupported types.

Reply:
FWIW, if you're serializing `DurableTaskTriggerMetrics` as JSON before inserting the properties into Azure Storage, then you can use serialization attributes to separate the class properties from the actual JSON data contract. For example, this is a potential workaround that I used to see the WCF team do all the time, and it works with Json.NET:

```csharp
[DataContract]
class DurableTaskTriggerMetrics : ScaleMetrics
{
    [DataMember(Name = "ControlQueueLatencies")]
    private string controlQueueLatenciesAsString;

    [DataMember]
    public int PartitionCount { get; set; }

    [DataMember]
    public string ControlQueueLengths { get; set; }

    public List<int> ControlQueueLatencies { get; set; }
    
    [OnSerializing]
    private void OnSerializing(StreamingContext ctx)
    {
        this.controlQueueLatenciesAsString = JsonConvert.SerializeObject(this.ControlQueueLatencies);
    }
    
    [OnDeserializing]
    private void OnDeserializing(StreamingContext ctx)
    {
        this.ControlQueueLatencies = JsonConvert.DeserializeObject<List<int>>(this.controlQueueLatenciesAsString);
    }
}
```